File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        res=[]
        for i, num in enumerate(nums):
            if num == target:
                res.append(i)

        if len(res) == 0:
            return [-1, -1]
        else:
            return [res[0], res[-1]]

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.rs
impl Solution {
    pub fn search_range(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut res = vec![];
        for (i, &n) in nums.iter().enumerate() {
            if n == target {
                res.push(i as i32);
            }
        }
        if res.is_empty() {
            vec![-1, -1]
        } else {
            vec![res[0], *res.last().unwrap()]
        }
    }
}